src/herder/HerderSCPDriver.cpp

```diff
@@ -319,18 +316,17 @@ HerderSCPDriver::validateValueHelper(uint64_t slotIndex, StellarValue const& b,
         return SCPDriver::kInvalidValue;
     }
 
-    if ((!nomination || lcl.ledgerVersion < 11) &&
-        b.ext.v() != STELLAR_VALUE_BASIC)
+    bool const expectSignedValue =
+        nomination ? (lcl.ledgerVersion >= 11)
```

Comment:
In this commit, you're missing enforcement that we have signed values all the time after protocol 14 (if `lcl.ledgerVersion >= 14` we expect everything to be signed).

I am not sure why some things are checked here and not at the top of the file (there is no point in returning "maybe valid" if there is no way values are valid because they're not signed)

Reply:
> In this commit, you're missing enforcement that we have signed values all the time after protocol 14 (if `lcl.ledgerVersion >= 14` we expect everything to be signed).

Why is that?  If `nomination` is `true`, then for protocols after `14`, `expectSignedValue` will be `true` because it's `true` for all protocols after `11`, and if `nomination` is `false` (i.e. we're in the ballot protocol), then for protocols after `14`, `expectedSignedValue` will be `true` because `compositeValueType()` will return `STELLAR_VALUE_SIGNED` for all such protocols.

> I am not sure why some things are checked here and not at the top of the file (there is no point in returning "maybe valid" if there is no way values are valid because they're not signed)

Is that not one of the cases that Marta was referring to in [this comment](https://github.com/stellar/stellar-core/pull/2625#discussion_r455438706)?